varname/ignore.py

```diff
@@ -156,87 +168,112 @@ def match(self, frame_no: int, frameinfos: List[inspect.FrameInfo]) -> bool:
     def __repr__(self):
         return f'<IgnoreDecorated({self.ignore[0].__name__}, {self.ignore[1]})>'
 
-class IgnoreQualname(IgnoreElem):
-    """Ignore a function by qualified name"""
-    def __init__(self,
-                 ignore: Tuple[Optional[Union[ModuleType, str]], str]) -> None:
+class IgnoreModuleQualname(IgnoreElem):
+    """Ignore calls by qualified name in the module"""
+    def __init__(self, ignore: Tuple[ModuleType, str]) -> None:
         super().__init__(ignore)
-        self.module_flag = (
-            None if not ignore[0]
-            else 'module' if isinstance(ignore[0], ModuleType)
-            else 'filename'
-        )
 
         # check uniqueness of qualname
         self.qname_checked = False
-        if self.module_flag == 'module':
-            modfile = getattr(self.ignore[0], '__file__', None)
-            if modfile and path.isfile(modfile):
-                source = Source.for_filename(modfile, self.ignore[0].__dict__)
+        modfile = getattr(self.ignore[0], '__file__', None)
+        if modfile and path.isfile(modfile):
+            source = Source.for_filename(modfile, self.ignore[0].__dict__)
```

Comment:
You should still check source.tree, and once you do, there's no need to check path.isfile.